packages/services/src/kernel/default.ts

```diff
@@ -1533,11 +1533,9 @@ namespace Private {
 
     // Handle the restart on all of the kernels with the same id.
     await Promise.all(
-      runningKernels.map(async k => {
-        if (k.id === kernel.id) {
-          await k.handleRestart();
-        }
-      })
+      runningKernels.map(
+        k => (k.id === kernel.id ? k.handleRestart() : Promise.resolve(void 0))
```

Comment:
How about: `runningKernels.filter(k => k.id === kernel.id).map(k => k.handleRestart())`

Reply:
Ooh, even better :smile:

Reply:
Haha I was just thinking of this @afshin 👍